pkg/discovery/discovery.go

```diff
@@ -124,7 +128,55 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) (errCount int) {
 		defer os.RemoveAll(outpath)
 	}
 	trackErrorsFor("assembling results tarball")(err)
+
+	// 9. Mark final annotation stating the results are available and status is completed.
+	trackErrorsFor("updating pod status")(
+		updateStatus(kubeClient, cfg.Namespace, pluginaggregation.CompleteStatus),
+	)
+
 	logrus.Infof("Results available at %v", tb)
 
 	return errCount
 }
+
+// updateStatus changes the summary status of the sonobuoy pod in order to
+// effect the finalized status the user sees. This does not change the status
+// of individual plugins.
+func updateStatus(client kubernetes.Interface, namespace string, s string) error {
+	pod, err := client.CoreV1().Pods(namespace).Get(pluginaggregation.StatusPodName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not retrieve sonobuoy pod")
+	}
+
+	if pod.Status.Phase != corev1.PodRunning {
```

Comment:
If the pod is running, why return an error?

Reply:
It's that it is _not_ running, so we don't want to check the annotations on the pod and want to give a more clear error.

Reply:
Ahh, you're right, sorry for the confusion there, I misread that.